### PR TITLE
Create base template

### PR DESCRIPTION
**Pull request by yankeexe**
**Things done:**

- starter template using Nextjs
- added prettier for formatting
- added prettier check to CI
- added editorconfig for cross text-editor formatting
- update readme with instructions.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    "Yankee Maharjan",
+    "Nikita Shrestha"
+  ],
+  "license": "ISC",
+  "bugs": {
+    "url": "https://github.com/leapfrogtechnology/resume-builder/issues"
+  },
+  "homepage": "https://github.com/leapfrogtechnology/resume-builder#readme",
+  "dependencies": {
+    "eslint-config-leapfrog": "^3.0.0",
```
Comment (sagunji) [inline package.json:28]: `eslint` should be added as dev dependency

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,8 @@
+.DS_*
+*.log
+logs
+**/*.backup.*
+**/*.back.*
+
+node_modules
+bower_components
```
Comment (sagunji) [inline .gitignore:8]: I think this was added by the starter kit. Right?

File: package.json (hunk 1 above)
Comment (yankeexe) [reply package.json:28]: huss, was looking into how to configure it. Will update in next commit.

File: .gitignore (hunk 2 above)
Comment (yankeexe) [reply .gitignore:8]: From https://gitignore.io/

Comment (sagunji) [reply .gitignore:8]: Yes. We won't be using bower ;)

File: .prettierrc [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+  "semi": true,
+  "tabWidth": 2,
+  "printWidth": 120,
+  "singleQuote": true
+}
```
Comment (leapfrog-saugat) [inline .prettierrc:6]: Use this config for prettier. This is from the reactjs-starter pack.
```
{
  "printWidth": 120,
  "tabWidth": 2,
  "useTabs": false,
  "semi": true,
  "singleQuote": true,
  "bracketSpacing": true,
  "arrowParens": "avoid"
}

```

File: .prettierignore [hunk 4]
```diff
@@ -0,0 +1,5 @@
+node_modules/
```
Comment (leapfrog-saugat) [inline .prettierignore:1]: Make sure to ignore the following if not already.

```
# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.

# dependencies
/node_modules
/.pnp
.pnp.js

# testing
/coverage

# production
/build

# misc
.DS_Store
.env
.env.local
.env.development.local
.env.test.local
.env.production.local

npm-debug.log*
yarn-debug.log*
yarn-error.log*

```

File: package.json [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    "next": "^9.3.5",
+    "react": "^16.13.1",
+    "react-dom": "^16.13.1"
+  },
+  "devDependencies": {
```
Comment (leapfrog-saugat) [inline package.json:33]: Add husky and lint-staged as well. Refer to the following package.json from reactjs-starter repo. This is still under construction though. eg Prettier is not placed in dev-dependencies. Also there are lots of other  packages which we won;t be using. Be sure to take care of it hai. 

> {
>   "name": "react-starter",
>   "version": "0.1.0",
>   "private": true,
>   "dependencies": {
>     "@emotion/core": "^10.0.28",
>     "@emotion/styled": "^10.0.27",
>     "axios": "^0.19.2",
>     "classnames": "^2.2.6",
>     "formik": "^2.1.4",
>     "husky": "^4.2.5",
>     "lint-staged": "^10.1.3",
>     "moment": "^2.24.0",
>     "node-sass": "^4.13.1",
>     "pinterpolate": "^0.2.1",
>     "prettier": "^2.0.4",
>     "react": "^16.13.1",
>     "react-datepicker": "^2.14.1",
>     "react-dom": "^16.13.1",
>     "react-redux": "^7.2.0",
>     "react-router-dom": "^5.1.2",
>     "react-scripts": "^3.4.1",
>     "react-table": "^7.0.4",
>     "redux": "^4.0.5",
>     "redux-persist": "^6.0.0",
>     "redux-thunk": "^2.3.0",
>     "yup": "^0.28.3"
>   },
>   "lint-staged": {
>     "src/**/*.{js,jsx,ts,tsx,json,css,scss,md}": [
>       "prettier --write",
>       "git add"
>     ]
>   },
>   "scripts": {
>     "start": "react-scripts start",
>     "build": "react-scripts build",
>     "test": "react-scripts test",
>     "eject": "react-scripts eject"
>   },
>   "husky": {
>     "hooks": {
>       "pre-commit": "lint-staged"
>     }
>   },
>   "eslintConfig": {
>     "extends": [
>       "eslint-config-leapfrog/react"
>     ]
>   },
>   "browserslist": {
>     "production": [
>       ">0.2%",
>       "not dead",
>       "not op_mini all"
>     ],
>     "development": [
>       "last 1 chrome version",
>       "last 1 firefox version",
>       "last 1 safari version"
>     ]
>   },
>   "devDependencies": {}
> }
>

File: package.json [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    "dev": "next",
+    "build": "next build",
+    "start": "next start",
+    "format": "prettier --write . '**/*.{ts,js,md,yaml,json}'",
+    "format:check": "prettier --check . '**/*.{ts,js,md,yaml,json}'",
+    "lint": "eslint pages"
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "npm run lint"
```
Comment (leapfrog-saugat) [inline package.json:16]: `"pre-commit": "lint-staged";`

```
"lint-staged": {
    "src/**/*.{js,jsx,ts,tsx,json,css,scss,md}": [
      "prettier --write",
      "git add"
    ]
  },
```

Comment (leapfrog-saugat) [reply package.json:16]: And let's name the folder `src` instead of `pages` for consistency.

**Pull request merged**